Redefinition of typedef 'OstreeContentWriter'

```
In file included from ../src/builder-git.c:34:
In file included from ../src/builder-flatpak-utils.h:29:
In file included from /usr/include/ostree-1/ostree.h:24:
/usr/include/ostree-1/ostree-content-writer.h:28:23: error: redefinition of typedef 'OstreeContentWriter' is a C11 feature [-Werror,-Wtypedef-redefinition]
   28 | G_DECLARE_FINAL_TYPE (OstreeContentWriter, ostree_content_writer, OSTREE, CONTENT_WRITER,
      |                       ^
/usr/include/ostree-1/ostree-types.h:39:37: note: previous definition is here
   39 | typedef struct _OstreeContentWriter OstreeContentWriter;
      |                                     ^
1 error generated.
[18/42] Compiling C object src/flatpak-builder.p/builder-flatpak-utils.c.o
FAILED: [code=1] src/flatpak-builder.p/builder-flatpak-utils.c.o 
clang -Isrc/flatpak-builder.p -Isrc -I../src -Isubprojects/libglnx -I../subprojects/libglnx -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/json-glib-1.0 -I/usr/include/x86_64-linux-gnu -I/usr/include/p11-kit-1 -I/usr/include/libxml2 -I/usr/include/ostree-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=gnu99 -O0 -g -Werror=empty-body -Werror=strict-prototypes -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=format=2 -Werror=format-security -Werror=format-nonliteral -Werror=pointer-arith -Werror=init-self -Werror=missing-declarations -Werror=return-type -Werror=overflow -Werror=int-conversion -Werror=parentheses -Werror=incompatible-pointer-types -Werror=misleading-indentation -Werror=missing-include-dirs -D_GNU_SOURCE -isystem /usr/include/mit-krb5 -isystem /usr/include/mit-krb5 -D_REENTRANT -pthread -isystem /usr/include/mit-krb5 -isystem /usr/include/mit-krb5 -MD -MQ src/flatpak-builder.p/builder-flatpak-utils.c.o -MF src/flatpak-builder.p/builder-flatpak-utils.c.o.d -o src/flatpak-builder.p/builder-flatpak-utils.c.o -c ../src/builder-flatpak-utils.c
In file included from ../src/builder-flatpak-utils.c:23:
In file included from ../src/builder-flatpak-utils.h:29:
In file included from /usr/include/ostree-1/ostree.h:24:
/usr/include/ostree-1/ostree-content-writer.h:28:23: error: redefinition of typedef 'OstreeContentWriter' is a C11 feature [-Werror,-Wtypedef-redefinition]
   28 | G_DECLARE_FINAL_TYPE (OstreeContentWriter, ostree_content_writer, OSTREE, CONTENT_WRITER,
      |                       ^
/usr/include/ostree-1/ostree-types.h:39:37: note: previous definition is here
   39 | typedef struct _OstreeContentWriter OstreeContentWriter;
      |                                     ^
1 error generated.
[19/42] Compiling C object src/flatpak-builder.p/builder-main.c.o
FAILED: [code=1] src/flatpak-builder.p/builder-main.c.o 
clang -Isrc/flatpak-builder.p -Isrc -I../src -Isubprojects/libglnx -I../subprojects/libglnx -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/sysprof-6 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gio-unix-2.0 -I/usr/include/json-glib-1.0 -I/usr/include/x86_64-linux-gnu -I/usr/include/p11-kit-1 -I/usr/include/libxml2 -I/usr/include/ostree-1 -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Werror -std=gnu99 -O0 -g -Werror=empty-body -Werror=strict-prototypes -Werror=missing-prototypes -Werror=implicit-function-declaration -Werror=format=2 -Werror=format-security -Werror=format-nonliteral -Werror=pointer-arith -Werror=init-self -Werror=missing-declarations -Werror=return-type -Werror=overflow -Werror=int-conversion -Werror=parentheses -Werror=incompatible-pointer-types -Werror=misleading-indentation -Werror=missing-include-dirs -D_GNU_SOURCE -isystem /usr/include/mit-krb5 -isystem /usr/include/mit-krb5 -D_REENTRANT -pthread -isystem /usr/include/mit-krb5 -isystem /usr/include/mit-krb5 -MD -MQ src/flatpak-builder.p/builder-main.c.o -MF src/flatpak-builder.p/builder-main.c.o.d -o src/flatpak-builder.p/builder-main.c.o -c ../src/builder-main.c
In file included from ../src/builder-main.c:30:
In file included from /usr/include/ostree-1/ostree.h:24:
/usr/include/ostree-1/ostree-content-writer.h:28:23: error: redefinition of typedef 'OstreeContentWriter' is a C11 feature [-Werror,-Wtypedef-redefinition]
   28 | G_DECLARE_FINAL_TYPE (OstreeContentWriter, ostree_content_writer, OSTREE, CONTENT_WRITER,
      |                       ^
/usr/include/ostree-1/ostree-types.h:39:37: note: previous definition is here
   39 | typedef struct _OstreeContentWriter OstreeContentWriter;
      |                                     ^
1 error generated.
```

ostree 2025.7
clang 21.1.8

--- a/configure.ac
+++ b/configure.ac
@@ -53,6 +53,12 @@ CC_CHECK_FLAGS_APPEND([WARN_CFLAGS], [CFLAGS], [\
   -Wstrict-aliasing=2 \
   -Werror=unused-result \
 ])])
+
+AS_CASE([$CC], [*clang*], [
+  CC_CHECK_FLAGS_APPEND([WARN_CFLAGS], [CFLAGS],
+                        [-Wno-typedef-redefinition])
+])
+
 AC_SUBST(WARN_CFLAGS)
 
 AC_ARG_ENABLE(sanitizers,

--- a/src/libostree/ostree-sysroot-deploy.c
+++ b/src/libostree/ostree-sysroot-deploy.c
@@ -399,8 +399,9 @@ again:
   if (!glnx_opendirat (new_etc_fd, path, TRUE, &target_dfd, error))
     return FALSE;
 
-  if (!dirfd_copy_attributes_and_xattrs (modified_etc_fd, path, src_dfd, target_dfd, flags,
-                                         cancellable, error))
+  if (!dirfd_copy_attributes_and_xattrs (modified_etc_fd, path, src_dfd, target_dfd,
+                                         sysroot_flags_to_copy_flags (0, flags), cancellable,
+                                         error))
     return FALSE;
 
   if (out_dfd)

